Reservation user story

ckaminer opened the pull request:
view commit messages for detailed breakdown of work done

ckaminer committed:
merge

ckaminer committed:
fix merge conflict

ckaminer committed:
initial changes
switched app layout to reflect user dropdown
updated factories
updated models to reflect schema changes
update relationship methods inside models

ckaminer committed:
finish setup for space index to show connection
update space index and show to reflect switch from socks to spaces
update sock presenter to be space presenter
update space show page to show space details
update space index page to show all spaces
update routes for space show to use slug instead of id
remove irrelevent content from space show page
set up slug for spaces (minus migration)
set up occupancy in space show page (hardcoded - still need to migrate)

ckaminer committed:
add test for user viewing dashboard
add to username dropdown in app layout file

rgbatty commented on app/helpers/application_helper.rb:
```diff
@@ -10,8 +10,7 @@ def present(object, klass = nil)
     if current_user
-      button_array << link_to("Logout", logout_path, method: :delete)
-      button_array << current_admin_display
+      # button_array << current_admin_display
```
Can we discuss this later? Looks like we are currently killing the button_array used for navigation. Sounds fine, but I want to make sure we all understand how this works going forward, or if we need to adjust to a different strategy. 😄

rgbatty commented on app/views/layouts/application.html.erb:
```diff
@@ -42,6 +42,13 @@
+              <li><%= link_to "Logout", "/logout", method: :delete %></li>
+              <li><%= link_to "Dashboard", "/dashboard" %></li>
+            </ul>
+          </li>
```
Of course, as I read further through the code, my previous comment is addressed. 😊 This looks good. 👍

rgbatty commented on app/views/spaces/index.html.erb:
```diff
@@ -0,0 +1,13 @@
+<div class="container socks row-well" id="all_socks">
```
Lets discuss layout changes for the site, and relevant custom CSS that will need to be removed.

rgbatty commented on spec/features/logged_in_user_can_see_dashboard_spec.rb:
```diff
@@ -0,0 +1,18 @@
+    click_on "My Trips"
+
+    expect(current_path).to eq orders_path
+  end
```
Lets make this test more robust by including specific details that should be included for a given Reservation. This may be subject to change depending on design choices for displaying these individual reservations.

For now, it'd be nice to see an expectation on the number of orders present.

Riizu commented:
This all looks quite good! 👍 

Let's discuss the topics you guys brought up in Slack (Slugging spaces, an occupancies column, and firstname/lastname columns) and the comments I made after lunch, and then merge. 😃

ckaminer committed:
update schema and factories
add slug and occupancy to spaces
change user name to first and last name
add order and user with order factories
update factories per schema changes
update feature tests to reflect schema and factory changes
feature tests passing
